packages/material-ui/src/Collapse/Collapse.js

```diff
@@ -198,20 +233,32 @@ const Collapse = React.forwardRef(function Collapse(props, ref) {
           className={clsx(
             classes.container,
             {
+              [classes.horizontal]: isHorizontal,
               [classes.entered]: state === 'entered',
-              [classes.hidden]: state === 'exited' && !inProp && collapsedHeight === '0px',
+              [classes.hidden]: state === 'exited' && !inProp && collapsedSize === '0px',
             },
             className,
           )}
           style={{
-            minHeight: collapsedHeight,
+            [isHorizontal ? 'minWidth' : 'minHeight']: collapsedSize,
             ...style,
           }}
           ref={handleRef}
           {...childProps}
         >
-          <div className={classes.wrapper} ref={wrapperRef}>
-            <div className={classes.wrapperInner}>{children}</div>
+          <div
+            className={clsx(classes.wrapper, {
+              [classes.horizontal]: isHorizontal,
+            })}
+            ref={wrapperRef}
+          >
+            <div
+              className={clsx(classes.wrapperInner, {
+                [classes.horizontal]: isHorizontal,
+              })}
```

Comment:
What about we use Slider's approach?

https://github.com/mui-org/material-ui/blob/65d036e8aab5e0fe9e3efbf788daa8af8db57757/packages/material-ui/src/Slider/Slider.js#L192

```suggestion
              className={classes.wrapperInner}
```